### ISSUE TITLE
[Feature Request] Commercial skip?

### ISSUE THREAD

**Issue by Yankees4life** [intro-skipper/intro-skipper#411]
Hello

Any plans to include commercial skip in the chapter skipping feature? I have some recorded shows on hand and I postprocess it with chapters that marks commercials so I think this would be a nice feature to have

**Comment by AbandonedCart** [comment]
Are there multiple commercial blocks?

**Comment by Yankees4life** [comment]
Yes

**Comment by dshanske** [comment]
Jellyfin Media Segments has a type for commercials. This could be extremely useful.

**Comment by AbandonedCart** [comment]
> Jellyfin Media Segments has a type for commercials. This could be extremely useful.

It doesn’t detect them. The easy part is skipping them. The challenge is finding an automated way to identify them.

**Comment by dshanske** [comment]
> > Jellyfin Media Segments has a type for commercials. This could be extremely useful.
> 
> It doesn’t detect them. The easy part is skipping them. The challenge is finding an automated way to identify them.

Yes, but this issue was about adding detection capability... I'll go with the simpler +1 on this idea then

**Comment by AbandonedCart** [comment]
> Yes, but this issue was about adding detection capability... I'll go with the simpler +1 on this idea then

I guess it wasn’t clear if you were saying the type was useful to getting this implemented or it would be useful to implement it because of the type.

**Comment by dshanske** [comment]
> > Yes, but this issue was about adding detection capability... I'll go with the simpler +1 on this idea then
> 
> I guess it wasn’t clear if you were saying the type was useful to getting this implemented or it would be useful to implement it because of the type. 

It would be useful to implement because the functionality to skip them is now built in

**Comment by AbandonedCart** [comment]
In the meantime, it is possible to manually set start and end points using segment editor. It’s not the most automated, but relatively simple.

**Comment by dshanske** [comment]
I was going to look to see about rigging a script to run comchap, which is a script around comskip that adds chapter segments to the file for commercials supposedly.

**Comment by AbandonedCart** [comment]
You'd still end up needing a way to turn that into something Jellyfin was able to read, unless it's just to fast forward manually

**Comment by jumoog** [comment]
I have already looked at `comskip` and how they do it. They use the ffmpeg library, whereas we need to use the command line ffmpeg CLI. If someone wants to port the `comskip` functionality to our plugin, I would accept that PR. In the meantime, use the segment editor.
